Mono.Cecil/Import.cs

```diff
@@ -491,12 +492,47 @@ public MetadataImporter (ModuleDefinition module)
 			this.module = module;
 		}
 
+		struct TypeRefKey : IEquatable<TypeRefKey>
+		{
+			string fullname;
+			string assembly;
+
+			public static TypeRefKey From (TypeReference r)
+			{
+				return new TypeRefKey { fullname = r.FullName, assembly = r.Module.FullyQualifiedName };
+			}
+
+			public override int GetHashCode ()
+			{
+				return fullname.GetHashCode () + assembly.GetHashCode ();
+			}
+
+			public bool Equals (TypeRefKey other)
+			{
+				return other.fullname == fullname && other.assembly == assembly;
+			}
+		}
+
 		TypeReference ImportType (TypeReference type, ImportGenericContext context)
 		{
 			if (type.IsTypeSpecification ())
 				return ImportTypeSpecification (type, context);
 
-			var reference = new TypeReference (
+			TypeReference reference;
+			var key = TypeRefKey.From (type);
+			if (cache.TryGetValue (key, out reference))
+			{
+				// Cecil only fills TypeRef GenericParameters if used ( bug ?)
```

Comment:
btw this should be properly fixed IMO, so I don't expect this to be merged as-is :smile: 

I didn't quite find where to make the proper fix though.
AFAICT Cecil creates trivial GenericParameter's only [upon use](https://github.com/jbevain/cecil/blob/master/Mono.Cecil/AssemblyReader.cs#L2755) during assembly read.